core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -1191,8 +1189,26 @@ object Infer {
                   }
                 }
           }
-        case _ => None
-      }
+        case _ =>
+          None
+      }).orElse(declared match {
+        case Type.Exists(vars, inT) =>
+          Type.instantiate(vars.iterator.toMap, inT, inferred, Map.empty).map {
+            case (_, subs) =>
+              validateSubs(subs.toList, left, right)
+                .as {
+                  new FunctionK[TypedExpr, TypedExpr] {
+                    def apply[A](te: TypedExpr[A]): TypedExpr[A] =
+                      // we apply the annotation here and let Normalization
+                      // instantiate. We could explicitly have
+                      // instantiation TypedExpr where you pass the variables to set
+                      TypedExpr.Annotation(te, declared)
+                  }
+                }
+          }
+        case _ =>
+          None
+      })
```

Comment:
We should be able to match on any Quantification and instantiate it correctly. The existential and universal require instantiating two different directions to only one or the other is easy.